controller/control.go

```diff
@@ -689,6 +724,7 @@ func (c *Controller) RemoveReplicaNoLock(address string) error {
 		}
 	}
 	c.UpdateVolStatus()
+	c.UpdateCheckpoint()
```

Comment:
Why do we need to update the checkpoint while removing replicas?

Reply:
Checkpoint is maintained at the controller to facilitate snapshot deletion at the replicas.
Before deleting an auto-generated snapshot, replica checks with the controller if checkpoint is set.
If checkpoint is set at the controller, it indicates that all the replicas are in sync, and this checkpoint is set at all the replicas.
Now any replicas can delete a snapshot below this checkpoint.
When a replica needs to be replaced, snapshot deletion process is the RW replicas need to be stopped. The only way to do it is by unsetting the checkpoint at the controller as soon as a replica goes down.